Src/FluentAssertions/Execution/AssertionScope.cs

```diff
@@ -395,6 +395,16 @@ public void Dispose()
                 parent.assertionStrategy.HandleFailure(failureMessage);
             }
 
+            IDictionary<string, object> reportables = contextData.GetReportable();
+
+            if (reportables.Count > 0)
+            {
+                foreach (KeyValuePair<string, object> reportable in reportables)
+                {
+                    parent.AddReportable(reportable.Key, reportable.Value.ToString());
+                }
+            }
```

Comment:
I haven't dug all the way into this to know all the implications, but we might be able to use `parent.contextData.Add(contextData);` instead.

This should also ensure that:
* `reportable.Value` is properly formatted with `Formatter.ToString` when `ToString()` is insufficient.
* `reportable.Value` is not formatted when the test passed.
* non-reportables are handled as well.
  That seems to benefit e.g. `NotBeEquivalentTo` to include the non-reportable section about the used equivalency settings.
  See `When_asserting_collections_not_to_be_equivalent_with_options_but_subject_collection_is_null_it_should_throw`

Reply:
may i ask what is the purpose of adding an AsserionScope here in: When_asserting_collections_not_to_be_equivalent_with_options_but_subject_collection_is_null_it_should_throw()
{
  Action act = () =>
  {
      using var _ = new AssertionScope();
      actual.Should().NotBeEquivalentTo(new[] { 1, 2, 3 }, opt => opt, "we want to test the failure {0}", "message");
  };
}

Reply:
It actually throws the exception, but with a different message:

    Expected exception message to match the equivalent of "Expected actual not to be equivalent *failure message*, but found <null>.", but "Expected actual not to be equivalent because we want to test the failure message, but found <null>.

With configuration:
- Use declared types and members
- Compare enums by value
- Compare tuples by their properties
- Compare anonymous types by their properties
- Compare records by their members
- Include non-browsable members
- Include all non-private properties
- Include all non-private fields
- Match member by name (or throw)
- Be strict about the order of items in byte arrays
- Without automatic conversion.
" does not.

Reply:
> May I ask what is the purpose of adding an AssertionScope here in:

As an assertion don't fail immediately when wrapped in an `AssertionScope`, we want to exercise the case, where even though we have null checked `Subject` we cannot safely assume it to be non-null afterwards.

This particular test was added in #1594.

Reply:
> It actually throws the exception, but with a different message:

Yes, and at first glimpse that extended failure message looks more correct to me, as it includes the "equivalency configuration".
The test should then be adapted to this, by adding a trailing `*` to the assertion.